src/main/java/core/basesyntax/supplier/FigureSupplier.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.supplier;
+
+import core.basesyntax.figure.Circle;
+import core.basesyntax.figure.Figure;
+import core.basesyntax.figure.IsoscelesTrapezoid;
+import core.basesyntax.figure.Rectangle;
+import core.basesyntax.figure.RightTriangle;
+import core.basesyntax.figure.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int randomNum = random.nextInt(1, 5);
+        switch (randomNum) {
+            case 1:
+                return new Circle(new ColorSupplier()
```

Comment:
There is no need to write a return in each case, you can create method-level variable and populate it in cases

Reply:
fixed